fs/ggml/ggml.go

```diff
@@ -104,6 +111,48 @@ func (kv KV) Bool(key string, defaultValue ...bool) bool {
 	return keyValue(kv, key, append(defaultValue, false)...)
 }
 
+func (kv KV) UintOrMaxArrayValue(key string, defaultValue uint32) uint32 {
+	_, max := kv.UintOrArrayValue(key, defaultValue)
+	return max
+}
+
+func (kv KV) UintOrMinArrayValue(key string, defaultValue uint32) uint32 {
+	min, _ := kv.UintOrArrayValue(key, defaultValue)
+	return min
+}
+
+func (kv KV) UintOrArrayValue(key string, defaultValue uint32) (uint32, uint32) {
+	if v, ok := keyValueUntyped(kv, key); ok {
+		switch v := v.(type) {
+		case *array:
+			length := len(v.values)
+			if length < v.size {
+				slog.Warn("trying to access a partially collected array. You probably want the whole array to be collected.", "key", key, "size", v.size, "len(values)", length)
+			}
+
+			if length == 0 {
+				return defaultValue, defaultValue
+			}
+			max := v.values[0].(int32)
+			min := v.values[0].(int32)
```

Comment:
This will panic on a non-int32 array

Reply:
ah yeah I did that to match the other parsing code in this file? Like <https://github.com/ollama/ollama/blob/424f648632c925ce14a75018c4dcab395e035993/fs/ggml/ggml.go#L95-L97>
I can make it more tolerant, but both examples we've seen in the wild are `int32` arrays and the rest of this parsing code is pretty strict if I'm understanding it correctly